src/App.js

```diff
@@ -1,16 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const startLikeCount = entries.filter((entry) => entry.liked).length;
+
+  const [likeCount, setLikeCount] = useState(startLikeCount);
+
+  const updateMessageData = (updatedMessage) => {
+    const messages = entries.map((message) => {
+      if (message.id === updatedMessage.id) {
+        if (updatedMessage.liked === true && message.liked === false) {
+          setLikeCount(likeCount + 1);
+        } else if (updatedMessage.liked === false && message.liked === true) {
+          setLikeCount(likeCount - 1);
+        }
+        return updatedMessage;
+      } else {
+        return message;
+      }
+    });
+
+    setEntries(messages);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between {entries[0].sender} and {entries[1].sender}
```

Comment:
Well done here. Just food for thought, what if the first two messages were from the same sender? How could this be modified? How could it be modified if there were more than two people in the chat? You don't need to make those changes, I just wanted to add some things to think about!